Modules/itertoolsmodule.c

```diff
@@ -897,27 +925,34 @@ tee_next(teeobject *to)
 }
 
 static int
-tee_traverse(teeobject *to, visitproc visit, void *arg)
+tee_traverse(PyObject *op, visitproc visit, void *arg)
 {
+    teeobject *to = teeobject_CAST(op);
     Py_VISIT(Py_TYPE(to));
-    Py_VISIT((PyObject *)to->dataobj);
+    Py_VISIT(to->dataobj);
```

Comment:
Don't remove this cast. `Py_VISIT` is documented to take `PyObject *`. In practice it works with anything and we're not likely to change that, but `PyObject *` is the correct type here.
```suggestion
    Py_VISIT((PyObject *)to->dataobj);
```

Reply:
Isn't it already a PyObject? (maybe not but I can't look at it now easily)